### PR TITLE
update and add case_joaat, partial_joaat, literal_joaat, tokened_joaat

### PR DESCRIPTION
**Pull request by PliskinDev**

### CODE REVIEW

File: rage/joaat.hpp [hunk 1]
```diff
@@ -25,4 +25,80 @@ namespace rage
...
+    }
 };
+
+
+
+
+
+
+
+
```
Comment (gir489returns) [inline rage/joaat.hpp:104]: Remove unnecessary whitespace.

**Review by gir489returns — CHANGES_REQUESTED** [review]

**Comment by gir489returns** [comment]
I've never heard of these. I know in RDO the JOAATs are case-sensitive due to the nature of the game being made later when JOAATs were used as actual identifiers and not just a way to save time for string comparators to squeeze more performance out of the dying 7th gen consoles it was launching. Can you provide usecases each function?

**Comment by PliskinDev** [comment]
> I've never heard of these. I know in RDO the JOAATs are case-sensitive due to the nature of the game being made later when JOAATs were used as actual identifiers and not just a way to save time for string comparators to squeeze more performance out of the dying 7th gen consoles it was launching. Can you provide usecases each function?

(case joaat) they use it in some tunables and files, (partial joaat) and (literal joaat) is used with some anim, vehicles, weapons, etc. It is the same as joaat, but without the (uint32_t initial_hash), as for the (tokened joaat) function, it is still new and has not been used yet, you can say that hash algorithms are based on joaat, but they are different and there are still many.
